### PR TITLE
DashboardList: Enable templating on search tag input

### LINKED ISSUE #29028: [FeatureRequest] DashboardList: Add templating on search tag input
**Steps to reproduce** / **Actual Result**:

Also searching (dashboard queries, tags) with template variables is not possible.

grafik

**Expected Result**
Also template variables should work for looking up dashboards.

**Environment**:
- Grafana version: 6.2.5
- Data source type & version: InfluxDB
- User OS & Browser: Windows 7, latest Firefox

> Not sure what is meant by `template variables should work for looking up dashboards` . To do this would require expanding all values of the template variables for every dashboard, which would be very time intensive. Or was there another idea?

This part of the issue has been fixed by @Maddin-619 

Now only the part for passing templating variables is missing.

_Originally posted by @florianwgnr in https://github.com/grafana/grafana/issues/18113#issuecomment-514505226_

### PR DESCRIPTION
**Pull request by delta50**
What this PR does / why we need it:

Now it is possible to pass variables to the search tag input field of the dashlist config. This is useful when you have a generic dashboard and would like to use the templating feature for showing only the relevant dashboards in the list depending on a variable value.

Which issue(s) this PR fixes:

Fixes: #29028
Relates #18113 #18167

Special notes for your reviewer:
This is an updated version of this outdated PR #18167

### CODE REVIEW

**Review by dprokop — CHANGES_REQUESTED** [review]
@delta50 thanks for the contribution. I've tested it and it seems like it works as expected.

Few things before we can merge this.
1. Docs needs to be updated (https://github.com/grafana/grafana/blob/abee02dbcd5fc3dae047d45180647f63bb279a29/docs/sources/panels/visualizations/dashboard-list-panel.md#L20)
2. Would be awesome if you could implement tests for this newly added feature
3. Your co-committer needs to sign CLA agreement (andrew.riedel@deepintent.com - I assume it's you) :)

Additionally- since the change does not seem to be a big one - could you also add the variables interpolation to the search query input?

File: docs/sources/panels/visualizations/dashboard-list-panel.md
Comment (achatterjee-grafana) [inline docs/sources/panels/visualizations/dashboard-list-panel.md:24]:
```suggestion
- **Search -** Display dashboards by search query or tags. You must enter at least one value in **Query** or **Tags**. For the **Query** and **Tags** fields, variable interpolation is supported using the dollar syntax, for example,`$my_var`.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added some copy edit suggestions, code changes will need a review from engineers.

File: public/app/plugins/panel/dashlist/DashList.tsx
Comment (torkelo) [inline public/app/plugins/panel/dashlist/DashList.tsx:40]: always pass it through replaceVars, there are multiple syntaxes so testing if the string contains variables is not needed

**Review by torkelo — CHANGES_REQUESTED** [review]

File: public/app/plugins/panel/dashlist/DashList.tsx
Comment (delta50) [reply public/app/plugins/panel/dashlist/DashList.tsx]: This is done.

**Comment by oddlittlebird** [comment]
This needs a milestone. Is it slated for 7.5?

**Comment by delta50** [comment]
I would very much appreciate if this was able to go out in 7.5 as it is will help simplify our grafana setup significantly. If there is anything left to do let me know and I can knock it out ASAP.

File: docs/sources/panels/visualizations/dashboard-list-panel.md
Comment (achatterjee-grafana) [inline docs/sources/panels/visualizations/dashboard-list-panel.md:24]:
```suggestion
- **Search -** Display dashboards by search query or tags. You must enter at least one value in **Query** or **Tags**. For the **Query** and **Tags** fields. Variable interpolation is supported, for example,`$my_var` or `[[my_var]]`.
```

**Review by achatterjee-grafana — COMMENTED** [review]
One minor copy-edit suggestion. Otherwise doc content looks Ok.

**Review by achatterjee-grafana — APPROVED** [review]

**Comment by delta50** [comment]
Hey @torkelo, is this good to go or are there more changes required?

**Comment by hugohaggmark** [comment]
> Hey @torkelo, is this good to go or are there more changes required?

Hi @delta50 I think you need to fix the build errors, seems to be prettier errors.

**Comment by delta50** [comment]
> > Hey @torkelo, is this good to go or are there more changes required?
> 
> Hi @delta50 I think you need to fix the build errors, seems to be prettier errors.

This is fixed, my yarn setup was not correct, so I was assuming these errors were due to something outside my PR.

**Review by torkelo — APPROVED** [review]

File: docs/sources/panels/visualizations/dashboard-list-panel.md
Comment (torkelo) [inline docs/sources/panels/visualizations/dashboard-list-panel.md:24]:
```suggestion
- **Search -** Display dashboards by search query or tags. You must enter at least one value in **Query** or **Tags**. For the **Query** and **Tags** fields. Variable interpolation is supported, for example,`$my_var` or `${my_var}`.
```

**Review by torkelo — APPROVED** [review]

**Pull request merged**
